packages/cubejs-firebolt-driver/src/FireboltDriver.ts

```diff
@@ -0,0 +1,221 @@
+
+  public constructor(config: Partial<FireboltDriverConfiguration> = {}) {
+    super(config);
+
+    this.config = {
+      readOnly: true,
+      ...config
+    };
+
+    this.connectionParams = {
+      username: <string>process.env.CUBEJS_DB_USER,
+      password: <string>process.env.CUBEJS_DB_PASS,
+      database: <string>process.env.CUBEJS_DB_NAME,
+      engineEndpoint: <string>process.env.CUBEJS_FIREBOLT_ENGINE_ENDPOINT,
+    };
+
+    const apiEndpoint = process.env.CUBEJS_FIREBOLT_API_ENDPOINT;
+
+    this.firebolt = Firebolt({
+      apiEndpoint,
+    });
+  }
+
+  public quoteIdentifier(identifier: string): string {
+    return `"${identifier}"`;
+  }
+
+  protected async initConnection() {
+    try {
+      const connection = await this.firebolt.connect(this.connectionParams);
+      return connection;
+    } catch (e) {
+      this.connection = null;
+      throw e;
+    }
+  }
+
+  public createTableSql(
+    quotedTableName: string,
+    columns: { name: string; type: string }[]
+  ) {
+    const cols = columns
+      .map(
+        (c) => `${this.quoteIdentifier(c.name)} ${this.fromGenericType(c.type)}`
+      )
+      .join(', ');
+
+    return `CREATE DIMENSION TABLE ${quotedTableName} (${cols})`;
+  }
+
+  public dropTable(tableName: string) {
+    if (tableName.match(/\./)) {
+      const [_, name] = tableName.split('.');
+      tableName = name;
+    }
+    return this.query(`DROP TABLE ${tableName}`, []);
+  }
+
+  protected async getConnection(): Promise<Connection> {
+    if (this.connection) {
+      const connection = await this.connection;
+      return connection;
+    }
+
+    this.connection = this.initConnection();
+    return this.connection;
+  }
+
+  public static dialectClass() {
+    return FireboltQuery;
+  }
+
+  public async createSchemaIfNotExists(_schemaName: string): Promise<any> {
+    // no-op
+  }
+
+  public async testConnection(): Promise<void> {
+    try {
+      await this.query('select 1');
+    } catch (error) {
+      throw new Error('Unable to connect');
+    }
+  }
+
+  public async query<R = Record<string, unknown>>(
+    query: string,
+    parameters?: unknown[]
+  ): Promise<R[]> {
+    const result = await this.queryResponse(query, parameters);
+    return result.data as R[];
+  }
+
+  public async stream(
+    query: string,
+    parameters: unknown[]
+  ): Promise<StreamTableData> {
+    const connection = await this.getConnection();
+
+    const statement = await connection.execute(query, {
+      settings: { output_format: OutputFormat.JSON },
+      parameters,
+    });
+
+    const { data: rowStream, meta: metaPromise } = await statement.streamResult();
+    const meta = await metaPromise;
+    const types = meta.map(({ type, name }) => ({ name, type: this.toGenericType(type) }));
+
+    return {
+      rowStream,
+      types,
+    };
+  }
+
+  public async unload(): Promise<DownloadTableCSVData> {
+    throw new Error('Unload is not supported');
```

Comment:
Does Fireboolt support UNLOADing data to s3/gs?

Reply:
I saw discussions about supporting unload few month ago. Not sure if it is ready. Need to research it